### PR TITLE
Update run details artefacts view to new table design

### PR DESCRIPTION
**Pull request by sophistifunk**
# Description

See [JENKINS-41833](https://issues.jenkins-ci.org/browse/JENKINS-41833).

# Submitter checklist
- [X] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by sophistifunk** [comment]
Waiting on CI, some unrelated Java tests were timing out locally :-/

**Comment by sophistifunk** [comment]
screen shot 2017-07-07 at 3 05 24 pm

**Comment by i386** [comment]
Noice 👍

**Comment by michaelneale** [comment]
error is lint (ignore the jenkins.io one): 

```
[INFO] /home/ubuntu/slave/workspace/NKINS-41833-artefacts-table-6OQCMVEM7KDNCMZSLU3IV4NKA57H6HNAYKFY2GWVMND4CVJWRLOQ/blueocean-dashboard/src/main/js/components/RunDetailsArtifacts.jsx
[INFO]    97:15  error  'style' is defined but never used                                                                                   no-unused-vars
[INFO]   119:62  error  The closing bracket must be aligned with the line containing the opening tag (expected column 25 on the next line)  react/jsx-closing-bracket-location
```

**Comment by sophistifunk** [comment]
Those are fixed in 7850b9385d410f00b405dea3f380dcc68431674e

**Comment by michaelneale** [comment]
ok - running aghain skipping the bad container

**Comment by michaelneale** [comment]
ah ok, that's better. 

@sophistifunk almost there - what is failing now is that there are a few places in the ATH that look for ye-olde artifacts tables - they look like they are all in log-karaoke: 

```
08:22:38 INFO  -  TEST FAILURE:  4 assertions failed, 1319 passed. (37m 27s)
08:22:38 INFO  - 
08:22:38 INFO  -  ??? log-karaoke/noStages
08:22:38 INFO  - 
08:22:38 INFO  -    - Step 09 (3.717s)
08:22:38 INFO  -    Failed [equal]: (0 == 1)  - expected "1" but got: "0"
08:22:38 INFO  -        at Object.<anonymous> (/home/ubuntu/slave/workspace/NKINS-41833-artefacts-table-6OQCMVEM7KDNCMZSLU3IV4NKA57H6HNAYKFY2GWVMND4CVJWRLOQ/acceptance-tests/src/test/js/log-karaoke/noStages.js:146:25)
08:22:38 INFO  - 
08:22:38 INFO  -  ??? log-karaoke/stages-block
08:22:38 INFO  - 
08:22:38 INFO  -    - Step 05 (24.11s)
08:22:38 INFO  -    Timed out while waiting for element <table.artifacts-table tr> to be present for 20000 milliseconds.  - expected "visible" but got: "not found"
08:22:38 INFO  -        at Object.module.exports.commands.validateNotEmptyArtifacts (/home/ubuntu/slave/workspace/NKINS-41833-artefacts-table-6OQCMVEM7KDNCMZSLU3IV4NKA57H6HNAYKFY2GWVMND4CVJWRLOQ/acceptance-tests/src/main/js/page_objects/blueocean/bluePipelineRunDetail.js:270:14)
08:22:38 INFO  -        at Object.module.exports.Step 05 (/home/ubuntu/slave/workspace/NKINS-41833-artefacts-table-6OQCMVEM7KDNCMZSLU3IV4NKA57H6HNAYKFY2GWVMND4CVJWRLOQ/acceptance-tests/src/test/js/log-karaoke/stages-block.js:77:14)
08:22:38 INFO  - 
08:22:38 INFO  -  ??? log-karaoke/stages-new
08:22:38 INFO  - 
08:22:38 INFO  -    - Step 05 (24.213s)
08:22:38 INFO  -    Timed out while waiting for element <table.artifacts-table tr> to be present for 20000 milliseconds.  - expected "visible" but got: "not found"
08:22:38 INFO  -        at Object.module.exports.commands.validateNotEmptyArtifacts (/home/ubuntu/slave/workspace/NKINS-41833-artefacts-table-6OQCMVEM7KDNCMZSLU3IV4NKA57H6HNAYKFY2GWVMND4CVJWRLOQ/acceptance-tests/src/main/js/page_objects/blueocean/bluePipelineRunDetail.js:270:14)
08:22:38 INFO  -        at Object.module.exports.Step 05 (/home/ubuntu/slave/workspace/NKINS-41833-artefacts-table-6OQCMVEM7KDNCMZSLU3IV4NKA57H6HNAYKFY2GWVMND4CVJWRLOQ/acceptance-tests/src/test/js/log-karaoke/stages-new.js:77:14)
08:22:38 INFO  - 
08:22:38 INFO  -  ??? log-karaoke/stages-old
08:22:38 INFO  - 
08:22:38 INFO  -    - Step 05 (24.31s)
08:22:38 INFO  -    Timed out while waiting for element <table.artifacts-table tr> to be present for 20000 milliseconds.  - expected "visible" but got: "not found"
08:22:38 INFO  -        at Object.module.exports.commands.validateNotEmptyArtifacts (/home/ubuntu/slave/workspace/NKINS-41833-artefacts-table-6OQCMVEM7KDNCMZSLU3IV4NKA57H6HNAYKFY2GWVMND4CVJWRLOQ/acceptance-tests/src/main/js/page_objects/blueocean/bluePipelineRunDetail.js:270:14)
08:22:38 INFO  -        at Object.module.exports.Step 05 (/home/ubuntu/slave/workspace/NKINS-41833-artefacts-table-6OQCMVEM7KDNCMZSLU3IV4NKA57H6HNAYKFY2GWVMND4CVJWRLOQ/acceptance-tests/src/test/js/log-karaoke/stages-old.js:82:14)
08:22:38 INFO  - 
```

so if you fire up ATH with ./run.sh --dev 
nightwatch ... log-karaoke - would show it up. 
Or look for 

```
table.artifacts-table tr
```

which I guess is the old? or is the new and not quite right? Hard to tell if there are ATH changes as it is late. It's cold and there are wolves after me. 

Next week.

**Comment by sophistifunk** [comment]
This is good, they're *actual* failures :)

File: blueocean-dashboard/src/main/js/components/RunDetailsArtifacts.jsx [hunk 1]
```diff
@@ -61,95 +61,97 @@ export default class RunDetailsArtifacts extends Component {
     }
 
     componentWillUnmount() {
-        this.artifacts = null;
+        this.artifactsPromisePromise = null;
```
Comment (kzantow) [inline blueocean-dashboard/src/main/js/components/RunDetailsArtifacts.jsx]: Typo here? should be `this.artifactsPromise` ?

Comment (sophistifunk) [inline blueocean-dashboard/src/main/js/components/RunDetailsArtifacts.jsx]: Yes. Not sure how I managed that. Churbrew :)

**Comment by michaelneale** [comment]
:bee:

**Pull request merged**